tests/system.rs

```diff
@@ -431,3 +433,33 @@ fn rustfmt_diff_no_diff_test() {
     let diff = make_diff("a\nb\nc\nd", "a\nb\nc\nd", 3);
     assert_eq!(diff, vec![]);
 }
+
+// Compare strings without distinguishing between CRLF and LF
+fn string_eq_ignore_newline_repr(left: &str, right: &str) -> bool {
+    let left = BytesIgnoreNewlineRepr(left.bytes().peekable());
+    let right = BytesIgnoreNewlineRepr(right.bytes().peekable());
+    left.eq(right)
+}
+
+struct BytesIgnoreNewlineRepr<'a>(Peekable<Bytes<'a>>);
+
+impl<'a> Iterator for BytesIgnoreNewlineRepr<'a> {
+    type Item = u8;
+    fn next(&mut self) -> Option<u8> {
+        self.0.next().map(|c| if c == b'\r' {
+            if *self.0.peek().unwrap_or(&0) == b'\n' {
+                self.0.next();
+                b'\n'
+            } else {
+                b'\r'
+            }
+        } else {
+            c
+        })
+    }
+}
+
+#[test]
+fn string_eq_ignore_newline_repr_test() {
+    assert!(string_eq_ignore_newline_repr("a\nb\nc\rd", "a\nb\r\nc\rd"));
```

Comment:
Could you add a few more test cases please? For example, where the left or right are empty, where they match trivially, where they don't match, and where the left and right are different lengths.